includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -1602,7 +1675,20 @@ public function is_amazon_pay_enabled() {
 	 * @return boolean
 	 */
 	public function is_link_enabled() {
-		return WC_Stripe_UPE_Payment_Method_Link::is_link_enabled( $this->gateway );
+		$is_enabled = WC_Stripe_UPE_Payment_Method_Link::is_link_enabled( $this->gateway );
+		if ( $this->is_product() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'link', 'product' );
+		}
+
+		if ( $this->is_cart() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'link', 'cart' );
+		}
+
+		if ( $this->is_checkout() ) {
+			return $is_enabled && $this->is_enabled_for_location( 'link', 'checkout' );
+		}
```

Comment:
non-blocking-nit: These checks are duplicated in `is_link_enabled`, `is_amazon_pay_enabled`, and `is_payment_request_enabled` buttons. Maybe there's an opportunity to refactor it.